service/r2/routes.go

```diff
@@ -40,7 +44,7 @@ func fetchNamespaces(s *service, _ *http.Request) (data interface{}, err error)
 
 func fetchNamespace(s *service, r *http.Request) (data interface{}, err error) {
 	vars := mux.Vars(r)
-	rs, err := s.store.FetchRuleSet(vars[namespaceIDVar])
+	rs, err := s.store.FetchRuleSetSnapshot(vars[namespaceIDVar])
```

Comment:
thats a great point. I think in this case this is a read-only endpoint so no one can mess with data in the database, and there is already some auth validation. But this is worth checking in on. I am not sure it is within the scope of this diff but Ill take a look and if it looks like something needs to be done here I can create a ticket.

Reply:
Yeah fair enough. Also somewhat related, there;s also the case that namespaceIDVar is empty and we should check if v, ok := vars[key], where ok is true? Not a big deal though since it'll just look up "" in the worst case and send back the appropriate error.

Reply:
Yeah I think personally I would leave the empty check as is. There is already checks in place further down the call stack that will detect this as a problem and raise the correct errors which will result if a 404. Not sure we need to add another check here.